### DISCUSSION TITLE
How to record information about uploaded files

### DISCUSSION THREAD

**Discussion by zeff-zou** [coder/code-server#6289] [Q&A]
I upload some files to the code server, how to record the information of these files. Does code server provide an upload event that can listen to?

For example, after I upload a file to code-server, I can record the information of this file to the database or other places by listening to this event.

**Comment by code-asher** [comment]
The only events available are through the VS Code extension API 
and currently it does not look like they offer anything for upload 
events.  The closest thing looks like `onDidCreateFiles` but that 
does not tell you whether it was updated or created another way 
unfortunately.

https://code.visualstudio.com/api/references/vscode-api#workspace
